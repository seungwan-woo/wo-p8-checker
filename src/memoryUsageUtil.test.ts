import { describe, expect, test } from "@jest/globals";
import {
  fetchMemoryUsageDocument,
  getMemoryUsageContent,
} from "./memoryUsageUtil";

describe("memoryUsageUtil", () => {
  test("should include 10MB/100MB, when fetchMemoryUsageDocument() and getMemoryUsageContent()", async () => {
    const memoryUsageDoc: Document | undefined =
      await fetchMemoryUsageDocument();
    expect(memoryUsageDoc).not.toBeUndefined();
    expect(memoryUsageDoc).toMatchSnapshot();

    const memoryUsage = getMemoryUsageContent(memoryUsageDoc!);
    expect(memoryUsage).not.toBeUndefined();
    expect(memoryUsage).toMatchSnapshot();
    expect(memoryUsage?.includes("100 MB")).toBeTruthy();
    expect(memoryUsage?.includes("10 MB")).toBeTruthy();
  });
});
