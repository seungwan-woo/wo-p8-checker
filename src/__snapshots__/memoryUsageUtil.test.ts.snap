// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`memoryUsageUtil should include 10MB/100MB, when fetchMemoryUsageDocument() and getMemoryUsageContent() 1`] = `
Document {
  "location": null,
}
`;

exports[`memoryUsageUtil should include 10MB/100MB, when fetchMemoryUsageDocument() and getMemoryUsageContent() 2`] = `
"Assets do not exceed the memory budget of 10 MB in ambient
            mode, and 100 MB in interactive mode."
`;
